File: Editorial/1427_Perform_String_Shifts/CPP/2_Compute_Net_Shift.cpp
class Solution {
public:
    string stringShift(string string, vector<vector<int>>& shift) {
        // Add up the left shifts and right shifts.
        vector<int> overallShifts(2, 0);
        for (auto& move : shift) {
            overallShifts[move[0]] += move[1];
        }
        int leftShifts = overallShifts[0];
        int rightShifts = overallShifts[1];

        // Determine which shift (if any) to perform.
        int len = string.length();
        if (leftShifts > rightShifts) {
            leftShifts = (leftShifts - rightShifts) % len;
            string = string.substr(leftShifts) + string.substr(0, leftShifts);
        } else if (rightShifts > leftShifts) {
            rightShifts = (rightShifts - leftShifts) % len;
            string = string.substr(len - rightShifts) +
                     string.substr(0, len - rightShifts);
        }

        return string;
    }
};

File: Editorial/1427_Perform_String_Shifts/Python3/2_Compute_Net_Shift.py
class Solution:
    def stringShift(self, string: str, shift: List[List[int]]) -> str:

        # Add up the left shifts and right shifts.
        overall_shifts = [0, 0]
        for direction, amount in shift:
            overall_shifts[direction] += amount
        left_shifts, right_shifts = overall_shifts

        # Determine which shift (if any) to perform.
        if left_shifts > right_shifts:
            left_shifts = (left_shifts - right_shifts) % len(string)
            string = string[left_shifts:] + string[:left_shifts]
        else:
            right_shifts = (right_shifts - left_shifts) % len(string)
            string = string[-right_shifts:] + string[:-right_shifts]

        return string
